paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
```

Comment:
In majel's code, when allocation fails, a bad allocation exception will be throw. 
```
if (ptr_ == nullptr) {
    throw std::bad_alloc();
}
```
If we do not throw an exception or CHECK FATAL, we have to check if the Array has been constructed correctly.
I suggest that we can return error state of most other operations except ```malloc/free``` and some CUDA operations. Because we can hardly do something to make the client to overcome if these system related APIs go down.